includes/class-amp-theme-support.php

```diff
@@ -303,20 +311,20 @@ public static function read_theme_support() {
 
 			$is_paired = ! empty( $args[ self::PAIRED_FLAG ] );
 
-			self::$support_added_via_theme = $is_paired ? self::TRANSITIONAL_MODE_SLUG : self::STANDARD_MODE_SLUG;
+			self::$support_added_via_theme  = $is_paired ? self::TRANSITIONAL_MODE_SLUG : self::STANDARD_MODE_SLUG;
+			self::$support_added_via_option = $theme_support_option;
 
-			/*
-			 * If the theme has transitional support, allow the user to opt for AMP-first mode via an option, since a theme
-			 * in transitional mode entails that it supports serving templates as both AMP and non-AMP, and this it is
-			 * able to serve AMP-first pages just as well as paired pages. Otherwise, consider that the the mode was
-			 * not set at all via option.
-			 */
-			self::$support_added_via_option = ( $is_paired && self::STANDARD_MODE_SLUG === $theme_support_option ) ? self::STANDARD_MODE_SLUG : null;
-			if ( self::STANDARD_MODE_SLUG === self::$support_added_via_option ) {
+			// Make sure the user option can override what the theme has specified.
```

Comment:
Good idea. Like you mentioned, the theme might have support, but plugins might not.